### ISSUE TITLE
CI - Missing Lint Artifacts (And CheckStyle)

### ISSUE THREAD

**Issue by ParaskP7** [wordpress-mobile/WordPress-FluxC-Android#2265]
### TL;DR

As part of [Gradle 7.3.3 & AGP 7.0.4](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2258) upgrade, I noticed that only the `plugins/woocommerce` module related Lint set of artifacts are being uploaded during a CI build (see `~/lint/lint-results-release.html/xml`). However, there are two more Lint related artifacts that need to be uploaded as well, for the `fluxc` and `example` modules.

PS: The same actually applies to CheckStyle related artifacts and the `tests` module.

### Details

This is most probably due to the way the `Save lint results` and `Uploading artifacts` works, which copies any Lint related artifacts that match the `.*/build/reports/.*` regex, thus overriding previous set of artifacts. This script probably expects that a repo would only contain one set of Lint related artifacts. But, for this repo we are actually interested in three sets of Lint related artifacts, for the below modules:

- `example` (App)
- `fluxc` (Library)
- `plugins/woocommerce` (Library)

Thus, and since the `plugins/woocommerce` probably runs last, this set of Lint artifacts overrides any preceding sets of Lint artifacts, leaving only those to be then uploaded.

PS: The same applies to the CheckStyle artifacts, which is actually worse as they are actually six sets of CheckStyle artifacts in total, for the below modules:

- `example` (App)
- `fluxc` (Library)
- `fluxc-annotations` (Java)
- `fluxc-processor` (Java)
- `plugins/woocommerce` (Library)
- `tests` (Java)

Thus, and since the `tests` probably runs last, this set of CheckStyle artifacts overrides any preceding sets of CheckStyle artifacts, leaving only those to be then uploaded.
